api/client/catalog/catalog.go

```diff
@@ -43,3 +57,47 @@ func IsCatalogAlreadyExistsError(err error) bool {
 	}
 	return apiErr.HTTPStatusCode == 400 && strings.Contains(apiErr.Message, "already exists")
 }
+
+// getAllResourcesWithPagination is a generic helper function that fetches all items from a paginated catalog API endpoint
+func getAllResourcesWithPagination[T any](
+	ctx context.Context,
+	client *client.Client,
```

Comment:
Should we change the type to an interface `client *client.Client` -> `client HTTPClient` ?